Battery_Checker.c

```diff
@@ -0,0 +1,91 @@
+#include <stdio.h>
+#include "Battery_Checker.h"
+
+BatteryStatus_tst BatteryStatus_st;
+
+float TempratureUnitConversion(float Temperature,TempratureUnit_ten TempUnit)
+{
+   float TempinCelcius = Temperature;
+  if(TempUnit == FAHRENHEIT)
+  {
+    TempinCelcius = ((Temperature - 32)*5/9);
+  }
+  return TempinCelcius;
+}
+int IsbatteryTempCheck(float Temprature)
+{
+   int ParamStatus = 0;
+  if(Temprature >= Min_Temp && Temprature <= Max_Temp) 
+  {
+      printf("Temperature out of range!\n");
+      if(Temprature < Min_Temp)
```

Comment:
this if cannot be true anytime because the temperature value is always >= min , as this is the condition to enter the block! did you mean to check if value is > max and < min ?